pkg/backplane/ca/authtest/testserver.go

```diff
@@ -0,0 +1,65 @@
+package authtest
+
+import (
+	"crypto/tls"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/heptio/sonobuoy/pkg/backplane/ca"
+)
+
+// Server is an extension of httptest.Server that uses our own CA
+type Server struct {
+	*httptest.Server
+	auth *ca.Authority
+	t    *testing.T
+}
+
+// NewServer is a passthrough wrapper for httptest.NewServer. It does not
+// use the CA at all. This provided only for debugging purposes.
+func NewServer(handle http.Handler, t *testing.T) *Server {
+	return &Server{
+		Server: httptest.NewServer(handle),
+		t:      t,
+	}
+}
+
+// NewTLSServer Wraps httptest.NewTLSServer, injecting our CA and TLS config
+func NewTLSServer(handle http.Handler, t *testing.T) *Server {
+	auth, err := ca.NewAuthority()
+	if err != nil {
+		t.Fatalf("Couldn't create certificate authority")
+	}
+	cfg, err := auth.MakeServerConfig("127.0.0.1")
```

Comment:
So is this just for the purposes of testing then?

Reply:
Yep. It's a drop-in replacement for [net/http/httptest](https://golang.org/pkg/net/http/httptest/)